code/player/TTTPlayer.AFK.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Sandbox;
+
+using TTTReborn.Settings;
+
+namespace TTTReborn.Player
+{
+    public partial class TTTPlayer
+    {
+        public readonly static List<InputButton> Buttons = Enum.GetValues(typeof(InputButton)).Cast<InputButton>().ToList();
+        private static int SecondsTillKick => ServerSettings.Instance.AFK.MinutesTillKick * 60;
+
+        private TimeSince _timeSinceLastAction = 0f;
+        private Rotation? _lastKnownRotation;
+
+        private void TickAFKSystem()
+        {
+            Client client = GetClientOwner();
+
+            if (client == null || client.IsBot)
+            {
+                return;
+            }
+
+            if (IsForcedSpectator || IsSpectator)
+            {
+                _timeSinceLastAction = 0;
+                return;
+            }
+
+            bool pressedAnyKeyPressed = Buttons.Any(button => Input.Pressed(button) || Input.Down(button));
```

Comment:
I think we can just do either `Pressed` or the `Down` check, no need for both?

Reply:
I rather check for both because the press is only on the action of down and released, but holding the button is not taking into account without Down.

Reply:
I'm just obsessing over performance that probably won't really matter, but I guess my idea is that we don't need an overly complex anti-AFK system. It would seem pretty easy to just periodically check just a few inputs from the player.

Thoughts @Alf21 

What you have is good though, basically checks every single possible input the player can make.

Reply:
I'd be willing to remove the rotation to keep Down.

Reply:
Let's get Alfs opinion on this, you could be correct with your approach here.

Reply:
Just `Input.Down`, more reliable (tested,  ref <https://github.com/Neoxult/Super-Terry-Brothers/blob/7356ce349977096d6053260541d9194249c6af87/code/ui/menu/Menu.cs#L45-L63>) and nearly same as `Pressed`, but continously

Reply:
So, no rotation either?

Reply:
Hmm... Based on the intention of the afk system. If it shall keep campers out as well, then without rotation. If it shall just check for afk, then with rotation (EyePos + API support, e.g. if a T is using a traitor item that interacts with another entity, not with the player pawn). Maybe, there is a way checking for mouse changes (that would avoid adding an API).

Reply:
I am sorry, this has gone over my head.

Reply:
Made the change to use only down.

Reply:
@Alf21 I believe what you are looking for is MouseDelta. I used it instead of Rotation. Good call.